tests/torch/quantization/test_quantizer_propagation_graph.py

```diff
@@ -1071,6 +1075,87 @@ def test_merge_redundant_subsequent_quantizers_across_graph(self, model_graph_qp
+                                              unified_scale_group_id_override=g_id)
+
+
+        paths_1 = get_edge_paths_for_propagation(qpsg,
+                                                 InsertionPointGraph.get_pre_hook_node_key('2 /C_0'),
+                                                 InsertionPointGraph.get_pre_hook_node_key('4 /E_0', input_port_id=0))
+
+        paths_2 = get_edge_paths_for_propagation(qpsg,
+                                                 InsertionPointGraph.get_pre_hook_node_key('3 /D_0'),
+                                                 InsertionPointGraph.get_pre_hook_node_key('4 /E_0', input_port_id=1))
+
+        qpsg.propagate_quantizer_via_path(pq_1, paths_1[0])
+        qpsg.propagate_quantizer_via_path(pq_2, paths_2[0])
+
+        merged_qp = qpsg.merge_quantizers_for_branching_node([pq_1, pq_2],
+                                                             [QuantizerConfig(per_channel=True)],
+                                                             [None, None],
+                                                             '1 /B_0')
+
+        return merged_qp[0]
+
+    @staticmethod
+    def get_model_graph_with_split_node():
+        mock_graph = nx.DiGraph()
+
+        #         (0 /A_0)
+        #            |
+        #         (1 /B_0)
+        #        /        \
+        #    (2 /C_0)   (3 /D_0)
+        #        \        /
+        #         (4 /E_0)
+
+        node_keys = ['A', 'B', 'C', 'D', 'E']
+        for node_key in node_keys:
+            mock_node_attrs = get_mock_nncf_node_attrs(op_name=node_key)
+            if node_key == "B":
+                # Split have no POST_HOOK
+                mock_node_attrs[NNCFGraph.NODE_TYPE_ATTR] = "split"
+            mock_graph.add_node(node_key, **mock_node_attrs)
+
+        mock_graph.add_edges_from([('A', 'B'), ('B', 'C'), ('B', 'D'), ('C', 'E'), ('D', 'E')])
+        mark_input_ports_lexicographically_based_on_input_node_key(mock_graph)
+        mock_graph = get_nncf_graph_from_mock_nx_graph(mock_graph)
+
+        ip_graph = get_ip_graph_for_test(mock_graph)
+        qpsg = QPSG(ip_graph)
+
+        operator_node_key_vs_trait_dict = {
+            '0 /A_0': QuantizationTrait.QUANTIZATION_AGNOSTIC,
+            '1 /B_0': QuantizationTrait.QUANTIZATION_AGNOSTIC,
+            '2 /C_0': QuantizationTrait.INPUTS_QUANTIZABLE,
+            '3 /D_0': QuantizationTrait.CONCAT,
+            '4 /E_0': QuantizationTrait.QUANTIZATION_AGNOSTIC,
+        }
+        model_graph_qpsg = TestQuantizerPropagationStateGraph.mark_nodes_with_traits(
+            qpsg, operator_node_key_vs_trait_dict
+        )
+        return model_graph_qpsg
+
+
+    def test_unifined_scale_type_for_split_node(self):
+
+        model_graph_qpsg = self.get_model_graph_with_split_node()
+        merged_pq = self.get_merged_pq(model_graph_qpsg)
+
+        assert merged_pq.unified_scale_type == UnifiedScaleType.UNIFY_ALWAYS
```

Comment:
You should also check that the resulting unified scale _group_ is correct - i.e. that there is only one group with a required number of PQs, or that there are no groups.